pkg/server/stream/v3/stream.go

```diff
@@ -137,3 +142,11 @@ func (s *StreamState) SetKnownResourceNamesAsList(url string, names []string) {
 	}
 	s.knownResourceNames[url] = m
 }
+
+func (s *StreamState) HasWarmingCluster() bool {
```

Comment:
Should this be more generic given that envoy also supports listener warming? Maybe `HasResourceWarming()`?

Reply:
Seems only Cluster warming needs to be considered, see below comments.